src/arch/arm/machine/gic_v3.c

```diff
@@ -130,10 +130,15 @@ static void gicv3_enable_sre(void)
     uint32_t val = 0;
 
     /* ICC_SRE_EL1 */
-    SYSTEM_READ_WORD(ICC_SRE_EL1, val);
-    val |= GICC_SRE_EL1_SRE;
+    SYSTEM_READ_WORD(ICC_SRE, val);
+    val |= GICC_SRE_SRE;
+    if (config_set(CONFIG_ARM_HYPERVISOR_SUPPORT)) {
+        /* don't trap to el2 on access to ICC_SRE_EL1 */
+        /* TODO is this what we want permanently @yanyan?? */
```

Comment:
@yyshen

Reply:
@yyshen could you have another look at this part?

Reply:
I think this is needed to avoid trapping EL1 ICC_SRE access to EL2.

Reply:
If we don't trap access to this R/W register, then it would also need to be saved and restored during context switch which we don't do.  I'm more of a fan of what's in the other PR, this init code still uses ICC_SRE_EL1 to enable system registers, but doesn't make it accessible to EL1 without trapping.  This way EL1 can still access system registers and doesn't need to access ICC_SRE_EL1 which we don't make available to it.